### ISSUE TITLE
Unable to parse URL: URL is not printable

### ISSUE THREAD

**Issue by ethus3h** [ludios/grab-site#11]
Getting a bunch of things like this in the wpull.log... is this normal? should I be worried that pages are missing from the WARC? Thanks :)

```
2015-07-25 04:50:02,400 - wpull.url - WARNING - Unable to parse URL ‘http://unicode.org/cldr/utility/confusables.jsp?&r=None&a=a\u200cb’: URL is not printable: 'http://unicode.org/cldr/utility/confusables.jsp?&r=None&a=a\u200cb'.
2015-07-25 04:50:02,429 - wpull.url - WARNING - Unable to parse URL ‘http://unicode.org/cldr/utility/confusables.jsp?&r=UTS46%2BUTS39&a=\u0646\u0627\u0645\u0647\u200c\u0627\u06cc.de’: URL is not printable: 'http://unicode.org/cldr/utility/confusables.jsp?&r=UTS46%2BUTS39&a=\u0646\u0627\u0645\u0647\u200c\u0627\u06cc.de'.
2015-07-25 04:50:02,452 - wpull.url - WARNING - Unable to parse URL ‘http://unicode.org/cldr/utility/confusables.jsp?&r=UTS46%2BUTS39&a=a\u200cb’: URL is not printable: 'http://unicode.org/cldr/utility/confusables.jsp?&r=UTS46%2BUTS39&a=a\u200cb'.
2015-07-25 04:50:02,483 - wpull.url - WARNING - Unable to parse URL ‘http://unicode.org/cldr/utility/confusables.jsp?&r=None&a=\u0646\u0627\u0645\u0647\u200c\u0627\u06cc.de’: URL is not printable: 'http://unicode.org/cldr/utility/confusables.jsp?&r=None&a=\u0646\u0627\u0645\u0647\u200c\u0627\u06cc.de'.
2015-07-25 04:50:02,488 - wpull.url - WARNING - Unable to parse URL ‘http://unicode.org/cldr/utility/confusables.jsp?&r=IDNA2003&a=a\u200cb’: URL is not printable: 'http://unicode.org/cldr/utility/confusables.jsp?&r=IDNA2003&a=a\u200cb'.
2015-07-25 04:50:02,506 - wpull.url - WARNING - Unable to parse URL ‘http://unicode.org/cldr/utility/confusables.jsp?&r=IDNA2003&a=\u0646\u0627\u0645\u0647\u200c\u0627\u06cc.de’: URL is not printable: 'http://unicode.org/cldr/utility/confusables.jsp?&r=IDNA2003&a=\u0646\u0627\u0645\u0647\u200c\u0627\u06cc.de'.
```

**Comment by ivan** [comment]
The relevant code in `wpull/url.py` is:

```py
    @classmethod
    @functools.lru_cache()
    def parse(cls, url, default_scheme='http', encoding='utf-8'):
        '''Parse a URL and return a URLInfo.'''
        url = url.strip()
        if not url.isprintable():
            raise ValueError('URL is not printable: {}'.format(ascii(url)))
```

So if you're seeing that message, I would suspect that the URLs *are* thrown away, unless they're linked-to elsewhere in a URL-encoded format.

**Comment by ivan** [comment]
cc @chfoo - should wpull urlencode these unicode URLs instead?

**Comment by chfoo** [comment]
@ivan: It should have accepted those but it looks like `str.isprintable()` is outdated or too specific. The idea was to discard unlikely URLs that contain unescaped control codes.

**Comment by ethus3h** [comment]
Wait, were the URLs discarded by instructing wpull to do so? Because without a specific setting being sent to wpull, that seems like a case of "the computer deciding it's smarter than the users" and deciding to discard URLs without being told so and with no way to override the setting.... Like, I think it's possible to prevent grab-site's global ignore set from happening by telling the crawl to start paused and then editing ./igsets. Is there not a flag like that for wpull?

**Comment by ivan** [comment]
This should be fixed as of 7c1afbefe0471e5a71059433f0d7b45c9f8d16ff / wpull 1.2.3
